src/ui/ResponsiveComponents/ResponsiveRecommendation.ts

```diff
@@ -92,6 +95,16 @@ export class ResponsiveRecommendation implements IResponsiveComponent {
   }
 
   private needSmallMode(): boolean {
+    if (this.searchInterface) {
+      switch (this.searchInterface.responsiveComponents.getResponsiveMode()) {
+        case 'small':
+          return true;
+        case 'auto':
```

Comment:
Should we do this instead ?
```
if (this.searchInterface.responsiveComponents.getResponsiveMode() == 'small') {
 return true
}
return this.coveoRoot.width() <= this.breakpoint;
```

Reply:
No, because you're missing on `medium` and `large`, that's why I putted a default.

Reply:
I am not a fan of breaking in a `switch` where the other cases `return`. Also, I think that using a defensive style to control flow is easier to read, simply because it reduces the need for nesting. I suggest inverting as in the following:

```
private needSmallMode(): boolean {
    const isWidthSmallerThanBreakpoint = this.coveoRoot.width() <= this.breakpoint;
    
    if (!this.searchInterface) {
        return isWidthSmallerThanBreakpoint;
    }
   
   switch(val) {
       case 1:
           return bool
      ...
      default:
         return bool
   }
}
```

Reply:
Done 😄